src/ray/object_manager/object_manager.cc

```diff
@@ -388,7 +388,7 @@ void ObjectManager::PushLocalObject(const ObjectID &object_id, const NodeID &nod
   auto local_chunk_reader = [this, object_id, total_data_size, metadata_size](
                                 uint64_t chunk_index,
                                 rpc::PushRequest &push_request) -> Status {
-    std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> chunk_status =
+    std::pair<const ObjectBufferPool::ChunkInfo, ray::Status> chunk_status =
```

Comment:
This is the first key issue: we need to copy the ChunkInfo since another thread may deallocate the buffer.